js/editor-ta.js

```diff
@@ -218,72 +246,134 @@ EditorTextArea.prototype.onChange = function() {
   });
 };
 
+EditorTextArea.prototype.getTotalHeight = function() {
+  return this.totalCalculatedHeight_;
+};
+
+EditorTextArea.prototype.getTotalWidth = function() {
+  if (this.settings_.get('wraplines')) {
+    return this.dimentions.width;
+  } else {
+    return this.characterDimentions_.width * this.longestLine_;
+  }
+}
+
+EditorTextArea.prototype.getLineHeight = function(line) {
+  const charWidth = this.characterDimentions_.width;
+  const charHeight = this.characterDimentions_.height;
+  if (!this.settings_.get('wraplines')) {
+    return charHeight;
+  }
+  // This line box filling algorithm defines how big a line will be by
+  // determining the points at which the line will break. This is at a
+  // whitespace character, in the middle of a word if it's the only word on the
+  // line or at a line break suggestion character (see below).
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens#Suggesting_line_break_opportunities
```

Comment:
so we're trying to match exactly the algorithm used by text area to decide where a line will break? seems prone to bugs. I can't think of a better way though. maybe add a comment here that states the assumption that the textarea wrap algorithm will match this exactly?